common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/RestClientVraNg.java

```diff
@@ -705,6 +742,25 @@ public List<VraNgCatalogItem> getCatalogItemsForProject(final String project) {
 		}
 	}
 
+	/**
+	 * Deletes a catalog item.
+	 *
+	 * @param catalogItemId catalog item id
+	 */
+	public void deleteCatalogItem(final String catalogItemId) {
+		try {
+			logger.info("Deleting catalog item with id '{}'", catalogItemId);
+			ResponseEntity<String> res = this.deleteCatalogItemPrimitive(catalogItemId);
+
+			if (!res.getStatusCode().is2xxSuccessful()) {
+				logger.error("Failed to delete catalog item with id '{}'", catalogItemId);
```

Comment:
I am not sure if it is worth doing but in theory we can extract a function for general delete - try ( execute passed function like a call back ) catch ( throw error for the specific passed type)

Reply:
I would do it, but I didn't get the idea 😄 can you add a bit more details? Or some pseudo code

Reply:
I am scared to do this 😄 

Jokes aside, Java is OOP language, doesn't play well with functions...

<img width="1049" alt="image" src="https://github.com/user-attachments/assets/bb9e159b-39a6-4a4d-bb32-972c720655b3">

Example ^^^ ....

Idk how to make it work

Reply:
The exception needs to be handled, even tho it will never happen as it's handled in the deleteItem..

Reply:
@joroaf any ideas on this one?